packages/manager/modules/pci/src/projects/project/notebooks/dashboard/dashboard.routing.js

```diff
@@ -20,6 +35,32 @@ export default /* @ngInject */ ($stateProvider) => {
             notebookId,
           },
         ),
+
+      killTasks: /* @ngInject */ (Poller) => (pattern) => Poller.kill(pattern),
+
+      needRefresh: /* @ngInject */ (notebook) => () =>
+        notebook.isStarting() || notebook.isStopping(),
+
+      waitNotebookToStartOrStop: /* @ngInject */ (
+        projectId,
+        notebook,
+        NotebookService,
+        Poller,
+      ) => () => {
+        const endPointUrl = NotebookService.buildGetNotebookUrl(
+          projectId,
+          notebook.id,
+        );
+        return Poller.poll(endPointUrl, null, {
+          interval: 2500,
+          successRule(notebookResponse) {
+            const n = new Notebook(notebookResponse, null);
```

Comment:
Name the variable as `response` to make it more readable ?

Reply:
Outdate.
notebook, i can't because there is another notebook variable.
I renamed it notebookModel.